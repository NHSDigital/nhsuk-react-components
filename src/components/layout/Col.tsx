import React, { FC, HTMLProps } from 'react';
import classNames from 'classnames';
import { ColWidth } from '@util/types/NHSUKTypes';

interface ColProps extends HTMLProps<HTMLDivElement> {
  width: ColWidth;
}

const Col: FC<ColProps> = ({ className, width, ...rest }) => (
  <div className={classNames(`nhsuk-grid-column-${width}`, className)} {...rest} />
);

export default Col;
