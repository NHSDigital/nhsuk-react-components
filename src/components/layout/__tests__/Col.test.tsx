import React from 'react';
import { render } from '@testing-library/react';
import Col from '../Col';

describe('Col', () => {
  it('matches snapshot', () => {
    const { container } = render(<Col width="full" />);

    expect(container).toMatchSnapshot('Col');
  });

  it('renders a grid column', () => {
    const { container } = render(<Col width="full" />);

    expect(container.querySelector('.nhsuk-grid-column-full')).toBeTruthy();
  });
});
