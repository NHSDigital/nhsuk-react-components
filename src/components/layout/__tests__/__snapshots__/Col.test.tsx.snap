// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col matches snapshot: Col 1`] = `
<div>
  <div
    class="nhsuk-grid-column-full"
  />
</div>
`;
